tools/build-bpf-skel.py

```diff
@@ -59,7 +59,16 @@ def gen_bpf_skeleton(bpftool_exec, in_file, out_fd):
 
     logging.debug('Generating BPF skeleton:')
     logging.debug('{}'.format(' '.join(bpftool_args)))
-    subprocess.check_call(bpftool_args, stdout=out_fd)
+    skel = subprocess.check_output(bpftool_args)
+    # libbpf is used via dlopen(), so rename symbols as defined
+    # in src/shared/bpf-dlopen.h
+    for s in (b'bpf_object__open_skeleton',
+              b'bpf_object__load_skeleton',
+              b'bpf_object__attach_skeleton',
+              b'bpf_object__detach_skeleton',
+              b'bpf_object__destroy_skeleton'):
+        skel = skel.replace(s, b'sym_' + s)
```

Comment:
Hmm, it doesn't matter too much, but operating on bytes here goes against the principle of decoding utf-8 early… Instead of operating on bytes, I think `check_output` should be called with `text=True` (or `universal_newlines=True`, if we need to supported older versions), and we should operate on normal text here.

And maybe instead of a loop, use regexps: `re.sub(r'bpf_(object__\w+_skeleton)', r'\1', skel)` ? We'd have to adjust the list here occasionally, and it'll get old quickly.

Reply:
Thanks, updated - the working regex line is slightly different: `skel = re.sub(r'(bpf_object__\w+_skeleton)', r'sym_\1', skel)`